### PR TITLE
chore: v6provider setup + ip resource and data source

### LINKED ISSUE #108: Convert `katapult_ip` resource and data sources to use Plugin Framework

### PR DESCRIPTION
**Pull request by ikadix**
Closes #108 

Adds a `v6provider` package
Adds the `ip` resource and data source to this package 
Add the `v6provider` to the muxserver

The `ip` resource in `provider` is kept and loaded as `katapult_legacy_ip` during acceptance tests.

The `provider` package is updated to no longer use `DefaultFunc` and no longer `required` fields. 
This is now handled in the configure step to align with the `v6provider` which cannot use `DefaultFunc`

### CODE REVIEW

File: internal/v6provider/v6provider.go [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+		resp.ResourceData = k.m
+		resp.DataSourceData = k.m
+		return
+	}
+
+	var conf KatapultProviderModel
+	diags := req.Config.Get(ctx, &conf)
+	resp.Diagnostics.Append(diags...)
+
+	m := &Meta{
```
Comment (jimeh) [inline internal/v6provider/v6provider.go:200]: Could we use the `NewMeta()` function here? Or is it more specifically tuned to the test setup?

File: docs/resources/ip.md [hunk 2]
```diff
@@ -33,10 +33,10 @@ resource "katapult_ip" "primary-db" {
 
 ### Optional
 
-- `label` (String) VIP label. Required when **vip** is `true`.
+- `label` (String) VIP label.Required when **vip** is `true`.
 - `network_id` (String)
-- `version` (Number) IPv4 or IPv6. Defaults to `4`.
-- `vip` (Boolean) Defaults to `false`.
+- `version` (Number) IPv4 or IPv6.
+- `vip` (Boolean)
```
Comment (jimeh) [inline docs/resources/ip.md:39]: The `version` and `vip` fields no longer mention the default value. Possibly we just need to add it to the description for each field in the schema?

**Review by jimeh — CHANGES_REQUESTED** [review]
Overall very good. Just a few minor tweaks and fixes mostly to ensure generated docs are accurate, along with some nitpicks.

File: docs/index.md [hunk 3]
```diff
@@ -31,15 +31,12 @@ provider "katapult" {
 <!-- schema generated by tfplugindocs -->
 ## Schema
 
-### Required
+### Optional
```
[the remark this answers is not in the document]
Comment (ikadix) [reply docs/index.md:34]: We could drop our support of env vars and provide guidance on how to use the env vars with input variables ? 
https://developer.hashicorp.com/terraform/language/values/variables#environment-variables

[the remark this answers is not in the document]
Comment (jimeh) [reply docs/index.md:34]: As discussed on Slack already, I'm happy for us to keep them marked as optional, cause technically they are as far as Terraform's config system is concerned if the values can come from custom environment variables. As long as we clarify that they're required in the property descriptions.

File: internal/provider/provider.go [hunk 4]
```diff
@@ -55,40 +55,38 @@ func New(c *Config) func() *schema.Provider { //nolint:funlen
...
-					Required:    true,
-					Sensitive:   true,
-					DefaultFunc: schema.EnvDefaultFunc("KATAPULT_API_KEY", nil),
-					Description: "API Key for Katapult Core API. Can be " +
+					Type:      schema.TypeString,
+					Optional:  true,
+					Sensitive: true,
+					Description: "**REQUIRED** via config or " +
+						"environment variable" +
+						"API Key for Katapult Core API. Can be " +
```
Comment (jimeh) [inline internal/provider/provider.go:63]: Some punctuation and spacing is needed after the required sentense :)

**Pull request merged**